server.js

```diff
@@ -3,36 +3,125 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+//API input
+import netflixTitles from './data/netflix-titles.json'
+
+
+const Key = 'mongodb+srv://ebbabw:gs4m4g4U6pC6ejLK@cluster0-yp7wy.mongodb.net/project-mongo?retryWrites=true&w=majority'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(Key, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Netflixdata = mongoose.model('Netflixdata', {
+
+  title: { type: String }, 
+  director: { type: String },
+  cast:  { type: String },
+  country: { type: String },
+  date_added: { type: String },
+  release_year: { type: Number },
+  rating: { type: String },
+  duration: { type: String },
+  listed_in: { type: String },
+  description: { type: String },
+  type: { type: String }
+})
+
+const Director = mongoose.model('Director', {
+
+  director: { type: String},
+
+})
+
+if (process.env.RESET_DATABASE) {
+  console.log("Resetting database...");
+  const seedDatabase = async () => {
+    await Netflixdata.deleteMany();
+    await Director.deleteMany();
+
+    netflixTitles.forEach((netflixdata) => new Netflixdata(netflixdata).save());
+    netflixTitles.forEach((netflixdata) => new Director(netflixdata).save());
+
+  };
+  seedDatabase();
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
 // Start defining your routes here
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hello! use these routes /netflixtitles (list all netflixtitles) /netflixtitles/directors (list all directors)')
 })
 
+app.get("/netflixtitles", async (req, res) => {
+  const { query } = req.query
+  const queryRegex = new RegExp(query, 'i');
+  const titles = await Netflixdata.find({title: queryRegex}).sort({
+    release_year: ""
+  })
+  res.json(titles.reverse());
+});
```

Comment:
Since you're only searching for title, maybe your query param could be called title instead of query?